File: Goland_Python3_Algorithm/Codeforce/888c.py

# --idea 
# -*- coding: utf-8 -*-
# @Author  : hakusai
# @Time    : 2023/12/11 15:56
from collections import defaultdict

if __name__ == '__main__':
    s = input()
    idx = defaultdict(list)
    for i, x in enumerate(s):
        idx[x].append(i)
    ans = float('inf')
    for a in idx.values():
        a = [-1] + a + [len(s)]
        maxD = 0
        for i in range(1, len(a)):
            maxD = max(maxD, a[i] - a[i - 1])
        ans = min(ans, maxD)
    print(ans)

File: Algorithm/Codeforce/888C.go
package main

import (
	"bufio"
	. "fmt"
	"os"
)

/*
   --idea
   -*- coding: utf-8 -*-
   @Author  : hakusai22
   @Time    : 2023/12/11 15:46
*/

func main() {
	s := ""
	Fscan(bufio.NewReader(os.Stdin), &s)
	pos := [26][]int{}
	for i, b := range s {
		b -= 'a'
		pos[b] = append(pos[b], i)
	}
	n := len(s)
	ans := n
	for _, ps := range pos {
		ps = append(append([]int{-1}, ps...), n)
		maxD := 0
		for i := 1; i < len(ps); i++ {
			maxD = max(maxD, ps[i]-ps[i-1])
		}
		ans = min(ans, maxD)
	}
	Fprint(os.Stdout, ans)
}

func min(a, b int) int {
	if b < a {
		return b
	}
	return a
}
func max(a, b int) int {
	if b > a {
		return b
	}
	return a
}
